gpt4all-backend/oscompat.cpp

```diff
@@ -0,0 +1,61 @@
+#include <filesystem>
+#include <string>
+
+#ifndef _WIN32
+#   include <dlfcn.h>
+#else
+#   include <algorithm>
+#   include <filesystem>
+#   include <sstream>
+#   define WIN32_LEAN_AND_MEAN
+#   ifndef NOMINMAX
+#       define NOMINMAX
+#   endif
+#   include <windows.h>
+#endif
+
+#include "oscompat.h"
```

Comment:
Can you explain why the include order here? <filesystem> is included in the header, but you include it first here so I imagine there is some reasoning behind this?